### PR TITLE
Syslog integration

### PR DESCRIPTION
**Pull request by sambhavsaggi**
~~This is 100% a work in progress, as I am much more comfortable with Rust than I am with Go. If anyone has any recommendation changes, feel free to send.~~ Ready to review!

### Pull Request Checklist

* [x] I have added any new tests that need to pass to `sytest-whitelist` as specified in [docs/sytest.md](https://github.com/matrix-org/dendrite/blob/master/docs/sytest.md)
* [x] Pull request includes a [sign off](https://github.com/matrix-org/dendrite/blob/master/docs/CONTRIBUTING.md#sign-off)

To use, switch the "file" logging parameter to "syslog" and use the following params:
```
address: localhost:514
protocol: tcp
```

Signed-off-by: Sambhav Saggi <samb@disroot.org>

### CODE REVIEW

**Comment by sambhavsaggi** [comment]
Alright, so I've tested this and yes, it does send information to syslog. I did this with a quick Debian container with rsyslog attacked and ports 514 forwarded with TCP.

**Comment by sambhavsaggi** [comment]
It doesn't look like UDP logging is working, so that's something to fix.

**Comment by sambhavsaggi** [comment]
It looks like TCP works and everything is alright, so I'm gonna take this out of draft

File: internal/log.go [hunk 1]
```diff
@@ -173,6 +178,34 @@ func setupFileHook(hook config.LogrusHook, level logrus.Level, componentName str
...
+	}
+
+}
+
+func setupSyslogHook(hook config.LogrusHook) {
+	syslogHook, err := lSyslog.NewSyslogHook(hook.Params["protocol"].(string), hook.Params["address"].(string), syslog.LOG_INFO, "")
+	if err == nil {
+		logrus.AddHook(syslogHook)
+	}
+}
```
Comment (S7evinK) [inline internal/log.go:207]: I think the hook should accept the same parameters as `setupFileHook` and use the `componentName` as the syslog tag. Also, maybe use the `logLevelHook`, like the file hook?

**Review by S7evinK — CHANGES_REQUESTED** [review]
Beside the comment, looks good.

File: internal/log.go (hunk 1 above)
Comment (sambhavsaggi) [reply internal/log.go]: Fixed!

**Review by S7evinK — APPROVED** [review]

**Review by kegsay — APPROVED** [review]
Thanks!

**Pull request merged**